src/main/java/hudson/plugins/git/GitPublisher.java

```diff
@@ -500,18 +514,34 @@ public void setEmptyTargetRepoToOrigin(){
     }
 
     public static final class BranchToPush extends PushConfig {
-        private String branchName;
 
-        public String getBranchName() {
-            return branchName;
-        }
+        private String srcBranchName;
+        private String branchName;   // dstBranchName
 
         @DataBoundConstructor
-        public BranchToPush(String targetRepoName, String branchName) {
+        public BranchToPush(String targetRepoName, String srcBranchName, String branchName) {
             super(targetRepoName);
+            this.srcBranchName = Util.fixEmptyAndTrim(srcBranchName);
             this.branchName = Util.fixEmptyAndTrim(branchName);
         }
 
+        @CheckForNull public String getSrcBranchName() {
+            return srcBranchName;
+        }
+
+        public void setSrcBranchName(String srcBranchName) {
+            this.srcBranchName = srcBranchName;
+        }
+
+
+        @CheckForNull public String getBranchName() {
```

Comment:
I would create <code>getDstBranchName()</code> and deprecate the current method. Javadoc would be also useful in any case